### DISCUSSION TITLE
Can I ask global route to fail early?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3346] [General]
@nerriav Global routing takes very little time for the project I am working on. However, when I make changes, these sometimes can make global routing impossible and I have to tweak my change.

When global routing fails, it doesnt fail quickly, it just runs "forever"(hours compared to minutes) before it ultimately fails.

Can I adjust a setting such that global route fails quickly so that I dont waste server build resources and I can quickly load the routing congestion report in the DRC viewer to drill down to the problem?

**Comment by rovinski** [comment]
In OR you can use the `-congestion_interations` argument which translates to `GLOBAL_ROUTE_ARGS=" -congestion_iterations X "` in ORFS. X is a positive integer. The default is 100.

**Reply by oharboe** [reply]
What should I look for in the log when choosing a new value?

Try with different values?

The only reference to itrations I found in the log:

```
[INFO GRT-0101] Running extra iterations to remove overflow.
[deleted]
Elapsed time: 2:22.98[h:]min:sec. CPU time: user 137.41 sys 5.50 (99%). Peak memory: 13555660KB.
```

So less than 3 minutes...

**Reply by maliberty** [reply]
Smaller values will be faster but try less hard to resolve the congestion.  You'll have to experiment to see the tradeoff.

**Reply by oharboe** [reply]
I see. Thanks!

**Comment by oharboe** [comment]
Syntax subtleties... `export GLOBAL_ROUTE_ARGS="-congestion_iterations 10 -verbose"`, failed with `[ERROR STA-0402] global_route -congestion_iterations 10 -verbose is not a known keyword or flag.`

`export GLOBAL_ROUTE_ARGS=-congestion_iterations 10 -verbose` worked...

It is still stuck on:

```
[INFO GRT-0101] Running extra iterations to remove overflow.
```

With `export GLOBAL_ROUTE_ARGS=-congestion_iterations 2 -verbose` it fails after 2:30 minutes.

I'll see how high I can make it until it "runs forever". If it fails after 10 minutes, then I have something I can try out. If there is nothing wrong, I would expect it to complete in ca. 3 minutes.

I'm curious if 10 iterations is enough for all the designs under test... https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1076
